File: stack/stack.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None

class Stack:
    def __init__(self, top=None):
        self.top = top
    def push(self, data):
        new_node = Node(data)
        new_node.next = self.top 
        self.top = new_node 

    def pop(self):
        if self.top is None:
            print("Stack is empty.")
            return None
        popped_node = self.top
        self.top = self.top.next 
        return popped_node.data 

    def peek(self):
        if self.top is None:
            print("Stack is empty.")
            return None
        return self.top.data 
    def display(self):
        current_node = self.top
        while current_node:
            print(current_node.data, end=",")
            current_node = current_node.next

stk=Stack()
for i in range(10):
    n=int(input("Enter values to push : "))
    stk.push(n)
print(stk.peek())
print(stk.pop())
stk.display()

File: stack.c
#include<stdio.h>

struct node{
    int data;
    struct node *link;
};

struct node *top=NULL;

void push(int x){
    struct node *newnode;
    newnode=(struct node*)malloc(sizeof(struct node));
    newnode->data=x;
    newnode->link=top;
    top=newnode;
}

void display(){
    struct node *temp;
    temp=top;
    if(temp==NULL){
        printf("Stack is empty");
    }
    else{
        while(temp!=NULL){
        printf("%d\n",temp->data);
        temp=temp->link;
    }
    }
}

void peek(){
    if(top==NULL){
        printf("Stack is Empty");
    }
    else{
        printf("Top element is %d",top->data);
    }
}

void pop(){
    struct node *temp;
    temp=top;
    if(top==NULL){
        printf("Stack is empty");
    }
    else{
        printf("The popped element is %d",top->data);
        top=top->link; // can also use temp->link as both are pointing to the same node
        free(temp);
    }
}